### PR TITLE
ISSUE-1199 Update DistributedLinagoraSecondaryBlobStoreTest to add eventdeadletters test

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java [hunk 1]
```diff
@@ -256,4 +267,39 @@ void sendEmailShouldResultingInEventuallySavingDataToBothObjectStoragesWhenSecon
...
             });
     }
+
+    @Test
+    void sendEmailShouldResultingInEventDeadLettersWhenSecondStorageIsDownForLongTime(GuiceJamesServer server) throws Exception {
+        secondaryS3.pause();
+
+        given()
+            .body(LinagoraEmailSendMethodContract$.MODULE$.bobSendsAMailToAndre(server))
+            .when()
```
Comment (vttranlina) [inline tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:277]: indent

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java [hunk 2]
```diff
@@ -256,4 +267,39 @@ void sendEmailShouldResultingInEventuallySavingDataToBothObjectStoragesWhenSecon
...
+            .post()
+            .then()
+            .statusCode(HttpStatus.SC_OK)
+            .contentType(JSON)
+            .extract()
+            .body()
+            .asString();
+
+        Thread.sleep(ONE_MINUTE);
```
Comment (vttranlina) [inline tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:286]: too long for test?

Comment (vttranlina) [reply tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:286]: is it necessary? I see `calmlyAwait.atMost(ONE_MINUTE)` above

Comment (hungphan227) [reply tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:286]: reduce calmlyAwait.atMost(ONE_MINUTE) to ten seconds

**Review by Arsnael — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java (hunk 2 above)
Comment (chibenwa) [reply tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:286]:
```suggestion
   calmlyAwait.atMost(ONE_MINUTE).untilAsserted(() -> {
       assertThat(blobStoreProbe.getEventDeadLetters().failedIds(group)
                    .collectList().block())
                    .isNotEmpty();
   };
```

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java [hunk 3]
```diff
@@ -256,4 +267,39 @@ void sendEmailShouldResultingInEventuallySavingDataToBothObjectStoragesWhenSecon
...
+                List<EventDeadLetters.InsertionId> insertionIds = blobStoreProbe.getEventDeadLetters().failedIds(group)
+                    .collectList().block();
+                List<FailedBlobEvents.BlobAddition> events = insertionIds.stream()
+                    .map(insertionId -> blobStoreProbe.getEventDeadLetters().failedEvent(group, insertionId).block())
+                    .map(FailedBlobEvents.BlobAddition.class::cast)
+                    .toList();
+                assertThat(bucketName).isEqualTo(events.getFirst().bucketName());
+                assertThat(events.stream().map(FailedBlobEvents.BlobAddition::blobId)).hasSameElementsAs(expectedBlobIds);
+            });
+    }
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:304]: No what I whish here is that we reprocess them (webadmin call) and ensure they get added in the second blob store.

**Comment by hungphan227** [comment]
ISSUE: Currently EventDeadLettersRedeliverService only use MailboxEventSerializer. Therefore, secondary blob store dead letters could not be loaded from cassandra db.

**Comment by hungphan227** [comment]
> ISSUE: Currently EventDeadLettersRedeliverService only use MailboxEventSerializer. Therefore, secondary blob store dead letters could not be deserialized after loaded from cassandra db.

ticket https://github.com/linagora/tmail-backend/issues/1253

**Review by chibenwa — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java (hunk 2 above)
Comment (quantranhong1999) [reply tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:279]: I see it is still ONE_MINUTE. Can we switch to await?

**Review by quantranhong1999 — APPROVED** [review]

File: tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java (hunk 2 above)
Comment (hungphan227) [reply tmail-backend/integration-tests/jmap/distributed-jmap-integration-tests/src/test/java/com/linagora/tmail/james/DistributedLinagoraSecondaryBlobStoreTest.java:279]: await for what?

**Comment by Arsnael** [comment]
Needs a rebase

**Comment by Arsnael** [comment]
Careful with that update james-sha1, a similar one from the bot has just been merged. Please double check :)

**Pull request merged**